lib/rubocop/cop/infinum/factory_bot_association.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Infinum
+      # This cop looks for 'association' in factories that specify a ':factory' option
+      #
+      # @example
+      #   #bad
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'Lord of the Rings' }
+      #        association :author
+      #      end
+      #    end
+      #
+      #    FactoryBoy.define do
+      #      factory :book do
+      #        title {'Lord of the Rings'}
+      #        author { association :author }
+      #      end
+      #    end
+      #
+      #    #good
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'Lord of the Rings' }
+      #        author { build(:author) }
+      #      end
+      #    end
+      #
+      #    FactoryBot.define do
+      #      factory :author do
+      #        name { 'Ivan Gundulic' }
+      #      end
+      #    end
+      class FactoryBotAssociation < ::RuboCop::Cop::Cop
+        MSG = 'Use %<association_name>s { build(:%<factory_name>s) } instead'
+
+        def_node_matcher :association_matcher, <<~PATTERN
```

Comment:
```suggestion
        def_node_matcher :association_definition, <<~PATTERN
```

I'd call this a definition, because the developer defines an association in the factory.

Reply:
True, thanks 😄